### PR TITLE
fix: broken layout hauki 618 and translate header and footer hauki 610

### PR DESCRIPTION
**Pull request by timwessman**
This PR includes fixes and some new wishes about multi language support. [HAUKI-610](https://helsinkisolutionoffice.atlassian.net/browse/HAUKI-610) & [HAUKI-618](https://helsinkisolutionoffice.atlassian.net/browse/HAUKI-618)

Description of commits:
- [fix: header layout](https://github.com/City-of-Helsinki/hauki-admin-ui/commit/3d0d21552bacc9600c546b6054a1c495e9b9a4f8)
  - Update to HDS 3.4.0 broke header's height in flex container. Disabling `flex-shrink` fixes that.
- [fix: show language selection on header. Select language from query param](https://github.com/City-of-Helsinki/hauki-admin-ui/commit/981790928e134f17f55bde2fe46a39f56c3de0c3)
  - Language selection didn't remain when page was changed or refreshed. So just adding language selection to url param and use that.
- [fix: checkbox width bug](https://github.com/City-of-Helsinki/hauki-admin-ui/commit/926b55ef3fe6ef3da227d71022f509cea12946b8)
  - There is a bug in HDS 3.4.0 (and 3.5.0) on checkbox width. Fix is coming to HDS soon, but now we fix this just by setting the width manually
- [fix: translate format functions and fix test mocks](https://github.com/City-of-Helsinki/hauki-admin-ui/commit/205d616d9c07224eddeb12af5a3a354722f7600a)
  - Just translate and fix tests. Mocking i18next for this non React module was but tricky, but it works this way
- [feat: translate header](https://github.com/City-of-Helsinki/hauki-admin-ui/commit/788be72764b2de5c3f6928f53bb091cf14d5b3c9)
  - Just translated header. Note: multi lang help -link doesn't exists (maybe coming some day). I didn't translate the name of the service `Aukiolot`
- [feat: translate footer](https://github.com/City-of-Helsinki/hauki-admin-ui/commit/4a433f2c0f3f634605ac67db69d47524d3592975)
  - Just translated footer. Translated also forgotten `close` button on header
- [fix: updated to hds 3.5.0 and fixed some typescript problems that cam…](https://github.com/City-of-Helsinki/hauki-admin-ui/commit/8a6219d5b2fdbbcd162c59d28b8cc20cdd97c909)
  - Updated to HDS 3.5.0. There is a fix for weird `crossOrigin` property bug. ..this is the easiest way to fix it. 

[HAUKI-610]: https://helsinkisolutionoffice.atlassian.net/browse/HAUKI-610?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[HAUKI-618]: https://helsinkisolutionoffice.atlassian.net/browse/HAUKI-618?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**HAUKI-ADMIN-UI branch is deployed to platta: https://hauki-pr378-admin-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://hauki-pr378-admin-ui.dev.hel.ninja :laughing::tada::tada::tada:

File: src/components/normal-opening-hours-validity/NormalOpeningHoursValidity.tsx [hunk 1]
```diff
@@ -72,6 +72,7 @@ const NormalOpeningHoursValidity = (): JSX.Element => {
                     openButtonAriaLabel="Valitse alkupäivämäärä"
                     required
                     value={startDateField.value}
+                    crossOrigin={undefined}
```
Comment (arpargo) [inline src/components/normal-opening-hours-validity/NormalOpeningHoursValidity.tsx:75]: What are these additions for? HDS documentation does not have these variables mentioned anywhere.

Comment (arpargo) [reply src/components/normal-opening-hours-validity/NormalOpeningHoursValidity.tsx:75]: Resolved with internal conversation, linked to https://github.com/creativetimofficial/material-tailwind/issues/427

File: src/components/footer/HaukiFooter.tsx [hunk 2]
```diff
@@ -1,34 +1,47 @@
...
+          logo={
+            <Logo
+              src={language === 'sv' ? logoSv : logoFi}
+              size="medium"
+              alt={t('Footer.LogoAlt')}
+            />
+          }
+          backToTopLabel={t('Footer.BackToTopLabel')}>
+          <Footer.Link
+            href="https://kaupunkialustana.hel.fi/aukiolosovelluksen-saavutettavuusseloste/"
```
[the remark this answers is not in the document]
Comment (timwessman) [reply src/components/footer/HaukiFooter.tsx]: Nyt korjattu. Linkit ei vielä johda mihinkään sisältöön, mutta nyt on paikka kaikille kielille.

**Review by arpargo — APPROVED** [review]
LGTM

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://hauki-pr378-admin-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**